front-end/src/store/transcriptUploadSlice.js

```diff
@@ -0,0 +1,20 @@
+import { createSlice } from '@reduxjs/toolkit'
+
+export const transcriptUploadSlice = createSlice({
+  name: 'transcriptUpload',
+  initialState: {
+    isUploaded: false
+  },
+  reducers: {
+    enableAnalysisButton: state => {
```

Comment:
These reducers should be renamed to something like "setTranscriptUploadStatus" or similar, instead of enable/disable analysis button. For 2 reasons: 1) the name implies that enabling the Analysis Button uploads a transcript, instead of the other way around which is actually what's happening. 2) it implies that the only reason we track transcript upload status is for the analysis button. This MIGHT be currently true, but I think it's better to keep it more general.

Also, it may be possible to change this into a single reducer function which takes as a parameter the boolean value, instead of having one for true and one for false. Up to you.